### PR TITLE
Fix remaining open issues

### LINKED ISSUE #24: Use doctrine/phpcr-odm, doctrine/orm as type names?
As with Sylius prefix with the vendor then the repository type preceded by a slash: `doctrine/phpcr-odm`, `puli/filesystem`

### PR DESCRIPTION
**Pull request by wouterj**
/fixes #24 and #19 
/ping @dantleech What do you think about the `phpcr/phpcr` type name for the PHPCR repository? It doesn't really use doctrine, so I thought the `phpcr` vendor prefix made more sense than the `doctrine` prefix.

### CODE REVIEW

File: src/DependencyInjection/Repository/Factory/DoctrinePhpcrOdmFactory.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ public function create(array $options)
      */
     public function getName()
     {
-        return 'doctrine_phpcr_odm';
+        return 'doctrine/phpcr-odm';
```
Comment (ElectricMaxxx) [inline src/DependencyInjection/Repository/Factory/DoctrinePhpcrOdmFactory.php]: again, that is the type name and won't become the repository name, which goes into the url when using resources-rest?

File: src/Registry/RepositoryRegistry.php [hunk 2]
```diff
@@ -73,7 +73,11 @@ public function all()
      */
     public function get($repositoryName = null)
     {
-        if (!isset($this->serviceMap[$repositoryName ?: $defaultRepositoryName])) {
+        if (null === $repositoryName) {
+            $repositoryName = $this->defaultRepositoryName;
```
Comment (ElectricMaxxx) [inline src/Registry/RepositoryRegistry.php]: Can't see it atm, but where does the `$defaultRepositoryName` come from?

File: src/DependencyInjection/Repository/Factory/DoctrinePhpcrOdmFactory.php (hunk 1 above)
Comment (wouterj) [inline src/DependencyInjection/Repository/Factory/DoctrinePhpcrOdmFactory.php]: Indeed. See the test config below for an example where this type name is used.

File: src/Registry/RepositoryRegistry.php (hunk 2 above)
Comment (wouterj) [inline src/Registry/RepositoryRegistry.php]: I pushed a commit to master today reintroducing the concept of a default repository. The `defaultRepositoryName` is passed to the constructor (and can be configured in the configuration).

**Pull request merged**